src/main.js

```diff
@@ -1,22 +1,81 @@
-var ideaArray = [];
+var ideaArray;
+var showStarred = false;
+var showSearched = false;
 
 var buttonSave = document.querySelector('#save-button');
+var buttonShowStar = document.querySelector('#show-star');
+var divOverlay = document.querySelector('.overlay');
 var ideaCards = document.querySelector('.idea-cards');
 var inputTitle = document.querySelector('#title-input');
 var inputBody = document.querySelector('#body-input');
 var imgHamburgerIcon = document.querySelectorAll('.drop-down-toggle');
 var sectionDefaultMenu = document.querySelector('.menu');
 var sectionDropDownMenu = document.querySelector('.drop-down');
-var divOverlay = document.querySelector('.overlay');
+var inputSearch = document.querySelector("#search-input");
+var buttonSearch = document.querySelector('#search-button')
 
 buttonSave.addEventListener('click', getUserIdea)
+buttonShowStar.addEventListener('click', function showOnlyStars() {
+  showStarred = !showStarred;
+  buttonShowStar.classList.toggle('active-starring')
+  displayIdeas();
+  toggleDropDown();
+})
 imgHamburgerIcon[0].addEventListener('click', toggleDropDown);
 imgHamburgerIcon[1].addEventListener('click', toggleDropDown);
 inputTitle.addEventListener('keyup', checkButtonStatus);
 inputBody.addEventListener('keyup', checkButtonStatus);
 ideaCards.addEventListener('click', checkClick);
+buttonSearch.addEventListener('click', toggleSearch);
+inputSearch.addEventListener('keyup', function updateDisplay() {
+  displayIdeas();
+});
+window.onload = doOnLoad;
+
+;function toggleSearch() {
+  showSearched = !showSearched;
+  buttonSearch.classList.toggle('active')
+  displayIdeas();
+}
 
-window.onload = checkButtonStatus;
+;function doOnLoad() {
+  ideaArray = localStorage.getItem('ideaArray')
+  ideaArray = JSON.parse(ideaArray);
+  if(ideaArray) {
+    for (var i = 0; i < ideaArray.length; i++) {
+      ideaArray[i] = new Idea(ideaArray[i].title, ideaArray[i].body, ideaArray[i].star, ideaArray[i].id)
+    }
+  } else {
+    ideaArray = [];
+  }
+  checkButtonStatus();
+  displayIdeas(true);
+}
+
+;function displayIdeas() {
+  var toDisplay = [];
+  if (!showStarred) {
+    toDisplay = ideaArray.concat();
+  } else {
+    for (var i = 0; i < ideaArray.length; i++) {
+      if(ideaArray[i].star) {
+        toDisplay.push(ideaArray[i])
+      }
+    }
+  } if(showSearched) {
+    for (var i = 0; i < toDisplay.length; i++) {
+      if(!(toDisplay[i].title.includes(inputSearch.value) || toDisplay[i].body.includes(inputSearch.value))) {
+        toDisplay.splice(i, 1);
+        i--;
+      }
+    }
```

Comment:
this function is a little long for me, can we go over it tomorrow before we begin?